src/test/streams_tests.cpp

```diff
@@ -329,12 +329,80 @@ BOOST_AUTO_TEST_CASE(streams_buffered_file)
     // by the rewind window (relative to our farthest read position, 40).
     BOOST_CHECK(bf.GetPos() <= 30);
 
+    bf.FindByte(31);
+    BOOST_CHECK_EQUAL(bf.GetPos(), 31U);
+
+    // If the byte isn't found, end of file.
+    try {
+        bf.FindByte(99);
+        BOOST_CHECK(false);
+    } catch (const std::exception& e) {
+        BOOST_CHECK(strstr(e.what(),
+                           "CBufferedFile::Fill: end of file") != nullptr);
+    }
```

Comment:
134de90f2b64ede6dbd812a9a88b00708685394d
```suggestion
    BOOST_CHECK_EXCEPTION(bf.FindByte(99), std::ios_base::failure, HasReason("CBufferedFile::Fill: end of file"));```